smpp/pdu/body.go

```diff
@@ -35,6 +35,9 @@ type Body interface {
 	// TLVFields return a decoded map of PDU TLV fields.
 	TLVFields() pdufield.TLVMap
 
+	// Set the optional parameters of PDU.
+	SetTLVFields(pdufield.TLVMap) error
```

Comment:
Setting TLVFields should work the same as Fields. Calling `TLVFields()` will return a map (which is a pointer in itself) so any changes to this TLVMap will be reflected inside the PDU.